### PR TITLE
test: add mocha/chai for functional testing

### PR DESCRIPTION
**Pull request by fippo**
this adds mocha and chai for functional testing.

Far from complete obviously, this is more a discussion starter. Had to make some changes to the chrome shim to make it work outside the browser environment.

### CODE REVIEW

**Comment by fippo** [comment]
Also add [sinon](http://sinonjs.org/) to the mix and then have something like this:
```
window.RTCPeerConnection = sinon.stub();
window.RTCPeerConnection.returns({
  addIceCandidate: sinon.stub().returns(Promise.resolve(mockStatsObject)),
});
```

**Comment by KaptenJansson** [comment]
Interesting that Mocha and Chai are very similar to [jasmine](https://github.com/webrtc/apprtc/blob/master/src/web_app/js/appcontroller_test.js) ;) I added that to the AppRTC [repo](https://github.com/webrtc/apprtc/blob/master/karma.conf.js) a little while back.

**Comment by fippo** [comment]
there is more than one js framework to do it ;-)

**Comment by KaptenJansson** [comment]
Mocha/Chai LGTM for me. Is there anything else you want to add or should I review as is?

**Comment by fippo** [comment]
go ahead :-)

File: package.json [hunk 1]
```diff
@@ -16,9 +16,11 @@
     "version": "",
     "postversion": "export GITTAG=\"echo $(git describe --abbrev=0 --tags | sed 's/^v//')\" && git push --force --set-upstream origin bumpVersion --follow-tags && git checkout gh-pages && git pull && cp out/adapter.js adapter.js && cp adapter.js adapter-`$GITTAG`.js && rm adapter-latest.js && ln -s adapter-`$GITTAG`.js adapter-latest.js && mkdir -p adapter-`$GITTAG`-variants && cp out/adapter.js adapter-`$GITTAG`-variants/ && cp out/adapter_*.js adapter-`$GITTAG`-variants/ && git add adapter.js adapter-latest.js adapter-`$GITTAG`.js adapter-`$GITTAG`-variants && git commit -m `$GITTAG` && git push --set-upstream origin gh-pages && git checkout master",
     "prepublish": "grunt build",
-    "test": "grunt && node test/run-tests.js"
+    "test": "grunt && mocha test/unit && node test/run-tests.js"
   },
   "dependencies": {
+    "chai": "^3.5.0",
+    "mocha": "^3.2.0",
```
Comment (KaptenJansson) [inline package.json]: Aren't these dev-dependencies?

File: src/js/chrome/chrome_shim.js [hunk 2]
```diff
@@ -107,21 +107,22 @@ var chromeShim = {
           pcConfig.iceTransports = pcConfig.iceTransportPolicy;
         }
 
-        return new webkitRTCPeerConnection(pcConfig, pcConstraints);
+        return new window.webkitRTCPeerConnection(pcConfig, pcConstraints);
```
Comment (KaptenJansson) [inline src/js/chrome/chrome_shim.js]: Maybe add a comment at the top of the shim about why we have to include 'window`?

File: test/unit/chrome.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const chai = require('chai');
```
Comment (KaptenJansson) [inline test/unit/chrome.js:9]: Add the webrtc copyright header with the current year.

File: test/unit/chrome.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+const expect = chai.expect;
```
Comment (KaptenJansson) [inline test/unit/chrome.js:10]: I wonder when we can go es6 all the way ;)

**Review by KaptenJansson — CHANGES_REQUESTED** [review]
Generally looking good, I also think we should update the test instructions at https://github.com/webrtc/adapter/blob/master/test/README.md

File: package.json (hunk 1 above)
Comment (fippo) [reply package.json]: done

File: src/js/chrome/chrome_shim.js (hunk 2 above)
Comment (fippo) [reply src/js/chrome/chrome_shim.js]: hrm. I was too lazy to figure out an excuse so I just set global.window = global ;-)

File: test/unit/chrome.js (hunk 3 above)
Comment (fippo) [reply test/unit/chrome.js:9]: done

**Review by fippo — COMMENTED** [review]
thanks. I've added two commits, can squash after re-review

**Comment by fippo** [comment]
hrm. Don't merge yet. I am modifying global.window in one test and this spills over to another. Need to find a way to reset this...

File: src/js/chrome/chrome_shim.js (hunk 2 above)
Comment (KaptenJansson) [reply src/js/chrome/chrome_shim.js]: Nice!

**Review by KaptenJansson — APPROVED** [review]

**Pull request merged**